src/apps/KintoAppRegistry.sol

```diff
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721EnumerableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import "../interfaces/IKintoID.sol";
+import "../interfaces/IKintoAppRegistry.sol";
+import "../interfaces/IKintoWalletFactory.sol";
+
+/**
+ * @title KintoAppRegistry
+ * @dev A contract that holds all the information of a KintoApp
+ */
+contract KintoAppRegistry is
+    Initializable,
+    ERC721Upgradeable,
+    ERC721EnumerableUpgradeable,
+    OwnableUpgradeable,
+    UUPSUpgradeable,
+    IKintoAppRegistry
+{
+    /* ============ Constants ============ */
+    uint256 public constant override RATE_LIMIT_PERIOD = 1 minutes;
+    uint256 public constant override RATE_LIMIT_THRESHOLD = 10;
+    uint256 public constant override GAS_LIMIT_PERIOD = 30 days;
+    uint256 public constant override GAS_LIMIT_THRESHOLD = 1e16; // 0.01 ETH
+
+    /* ============ State Variables ============ */
+
+    uint256 public override appCount;
+    mapping(address => IKintoAppRegistry.Metadata) private _appMetadata;
+    mapping(address => mapping(address => bool)) private _appSponsoredContracts; // other contracts to be sponsored
```

Comment:
```suggestion
    mapping(address => mapping(address => bool)) private _sponsoredContracts; // other contracts to be sponsored
```

> // other contracts to be sponsored
why is it "other"? I think i'm still trying to understand the difference between the `childContracts` and the `appSponsoredContracts` :/

Reply:
contracts that I am only willing to pay for it if they are part of my batch